crates/nu-command/src/filesystem/ucp.rs

```diff
@@ -0,0 +1,709 @@
+use nu_engine::env::current_dir;
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::ast::{Argument, Expr};
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, Span, Spanned, SyntaxShape, Type,
+};
+
+use std::path::PathBuf;
+use uucore::mods::backup_control::BackupMode;
+use uucore::mods::update_control::UpdateMode;
+
+const GLOB_PARAMS: nu_glob::MatchOptions = nu_glob::MatchOptions {
+    case_sensitive: true,
+    require_literal_separator: false,
+    require_literal_leading_dot: false,
+    recursive_match_hidden_dir: true,
+};
+
+/// Returns tuple of (Source paths, Target)
+pub fn parse_path_args(
```

Comment:
I'd say we're being too compatible here. I think the way to go is to make it super simple: parse the paths, pop the last one and check that both are non-empty. We can then ignore `--target-dir`